### PR TITLE
Warn unused and clamp

### PR DESCRIPTION
**Pull request by necessarily-equal**

### CODE REVIEW

**Comment by slipher** [comment]
I would rather keep `a = Clamp(a,b,c)`. I think ClampArg just makes it harder to read.

Other commits look good.

**Comment by bmorel** [comment]
I agree with slipher: the only reason I erroneously used Math::Clamp() without reading return value was because this is what unvanquished's API usually does, especially regarding vec3_t. I prefer, by far, to have return value.

**Comment by necessarily-equal** [comment]
I've dropped the ClampArg thing.

**Comment by bmorel** [comment]
LGTM

File: src/common/Math.h [hunk 1]
```diff
@@ -36,7 +36,8 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace Math {
 
     // Returns min if value is NaN
-    template<typename T> T Clamp(T value, T min, T max)
+    template<typename T> inline WARN_UNUSED_RESULT
```
Comment (slipher) [inline src/common/Math.h:39]: FYI templates are automatically inline.

**Review by slipher — COMMENTED** [review]
Any reason not to merge this?

**Comment by necessarily-equal** [comment]
You can merge it, I've just not connected to github for a least a week now

Le 1 février 2022 01:46:05 GMT+01:00, slipher ***@***.***> a écrit :
***@***.*** commented on this pull request.

**Pull request merged**